pysnmp/proto/secmod/eso/priv/aesbase.py

```diff
@@ -25,19 +26,47 @@ class AbstractAes(aes.Aes):
     keySize = 0
 
     # 3.1.2.1
+    # def localizeKey(self, authProtocol, privKey, snmpEngineID):
+    #     if authProtocol == hmacmd5.HmacMd5.serviceID:
+    #         localPrivKey = localkey.localizeKeyMD5(privKey, snmpEngineID)
+    #         for count in range(1, int(ceil(self.keySize * 1.0 / len(localPrivKey)))):
+    #             # noinspection PyDeprecation,PyCallingNonCallable
+    #             localPrivKey += md5(localPrivKey).digest()
+    #     elif authProtocol == hmacsha.HmacSha.serviceID:
+    #         localPrivKey = localkey.localizeKeySHA(privKey, snmpEngineID)
+    #         # RFC mentions this algo generates 480bit key, but only up to 256 bits are used
+    #         for count in range(1, int(ceil(self.keySize * 1.0 / len(localPrivKey)))):
+    #             localPrivKey += sha1(localPrivKey).digest()
+    #     else:
+    #         raise error.ProtocolError(
+    #             'Unknown auth protocol %s' % (authProtocol,)
+    #         )
+    #     return localPrivKey[:self.keySize]
+
+
+    #Cisco devices do not use https://tools.itef.org/pdf/draft_bluementhal-aes-usm-04.txt for key localization
+    #instead, they use the procedure for 3DES key localization specified in
+    #https://tools.itef.org/pdf/draft_reeder_snmpv3-usm-3desede-00.pdf
+    #the difference between the two is that the reeder draft does key extension by repeating the steps
+    #in the password to key alogorithm (hash phrase, then localize with SNMPEngine ID)
+    #Should this library support both key localization methods by some mechanism?
+    #Are there non-cisco devices out there that support the key localization for AES192/AES256 in https://tools.itef.org/pdf/draft_bluementhal-aes-usm-04.txt?
+    #Pysnmp Maintainers can decide this.
```

Comment:
I'd like to propose supporting both Blumenthal and Reeder schemas. The rationale is that it is very easy to do in pysnmp while neither of these algorithms reached the level of a single standard.

With pysnmp design, key localization code can be isolated in a dedicated class (e.g. `Aes256`) identified by SNMP protocol ID (e.g. `Aes256.serviceID`). So may be we could keep both versions of key localization in distinct classes identified by different OIDs (one could be as it is in RFC while the other with .1 suffix) so that users would have control over that via `pysnmp.entity.config.usm*` constants?